### ISSUE TITLE
Add eslint rules to catch issues flagged by SonarQube

### ISSUE THREAD

**Issue by thornbill** [jellyfin/jellyfin-web#3494]
Below is a breakdown of eslint rules that we should add to prevent things that SonarQube considers code smells by their priority in SonarQube.

These rules should be applied and fixed one per PR.

### Critical
* [x] [default-case-last](https://eslint.org/docs/rules/default-case-last) &mdash; #3493 
* [x] [no-void](https://eslint.org/docs/rules/no-void) &mdash; #3509 

### Major
* [x] [no-redeclare](https://eslint.org/docs/rules/no-redeclare) (already included by `eslint:recommended`)
* [x] [no-sequences](https://eslint.org/docs/rules/no-sequences) &mdash; #3609
* [x] [no-shadow](https://eslint.org/docs/rules/no-shadow) &mdash; #4058
* [x] [no-nested-ternary](https://eslint.org/docs/rules/no-nested-ternary) &mdash; #3603
* [ ] [max-params](https://eslint.org/docs/rules/max-params) with a max of 7 &mdash; #4035
* [x] [no-throw-literal](https://eslint.org/docs/rules/no-throw-literal) &mdash; #3798

### Minor
* [x] [no-return-await](https://eslint.org/docs/rules/no-return-await) &mdash; #3796
* [x] [no-unused-vars](https://eslint.org/docs/rules/no-unused-vars)
* [x] [no-duplicate-imports](https://eslint.org/docs/rules/no-duplicate-imports) &mdash; #3539

**Comment by sancodes** [comment]
I would like to help, if this is still open.

**Comment by thornbill** [comment]
Go for it. I only did the first one from the list. 🙂

**Comment by sancodes** [comment]
@thornbill I've added all the rules. Please let me know if I've missed anything. Thank you

**Comment by sancodes** [comment]
@thornbill I have added lint for `no-void`. Please help me review it. If any errors, I'm willing to fix it.

**Comment by sancodes** [comment]
@thornbill thank you for approving changes, if other issues are still open, I can try to tackle it. Let me know.

**Comment by thornbill** [comment]
@sancodes they are still open. I have been updating the list at the top of the issue, so it should be accurate. :slightly_smiling_face:

**Comment by sancodes** [comment]
@thornbill I will start with `major` section. I will open new PR through new branch.

**Comment by ryan-hartzell** [comment]
@thornbill I wanted to help out on these. I added a PR for `no-nested-ternary`. I'm not sure of the logic on all the fixes.

**Comment by sancodes** [comment]
@thornbill can you help review the changes so far, and merge it. Thank you.

**Comment by Stromwerk** [comment]
I would like to give this a go.

I decided to start with no-unused-vars, which is turned on by default (and as I suspected it's also already on in the Jellyfin code base). So a quick search for `no-unsed-vars` gave me all the instances which would need to be removed.
![image](https://user-images.githubusercontent.com/11208509/183498056-09c29dc5-1b75-4545-9896-005f0ff58334.png)

If my understanding is correct then specifically for `no-unused-vars` the criteria in this issue is met - the rule is on and although there are instances where it is disabled, there is a reason for it to be disabled.

The only fix I see is in the `fetchhelper.js` component, as the unused var is an `_` which can simply be removed. I will submit a PR with only that change.
